TACC/Core-Styles
# task/tup-534 Adds in media query for tables

## Overview

Added in mobile view for tables. Columns now go from horizontal to vertical in mobile view. 
Replaces the [old solution](https://github.com/TACC/Core-Styles/blob/v2.17.1/src/lib/_imports/elements/table--basic.css#L64-L71) (in Core-Styles) (and any code mentioning APCD like it does) with the [new solution](https://github.com/TACC/Core-CMS-Custom/blob/3bec84b/apcd-cms/src/apps/view_users/static/view_users_table/css/table.css#L65-L129) (from APCD).

## Related

- [TUP-534](https://tacc-main.atlassian.net/browse/TUP-534)

## Changes
(Most changes were done in table--basic. Does this need to happen elsewhere? @wesleyboar)
- Added a media query to handle the table being on a smaller screen
- Removed and added styles that pertain to different tables (Header, No Header, Paragraphs, Paragraphs with Headers, Nested Tables)

## Testing

1. Check for borders in correct spaces
2. Check that all data is available and able to be viewed
3. Watch for unwanted side-effects (missing data, unwanted padding, unwanted margin, borders in wrong spots)
4. This is the start to a mobile solution that WILL be iterated on, doesn't have to be perfect (yet), but should have all of the ingredients. 

## UI

| Web | Mobile |
| --- | --- |
| <img width="1011" alt="Screenshot 2024-01-25 at 11 41 59 AM" src="https://github.com/TACC/Core-Styles/assets/63771558/ac0ec0bf-7ac4-486f-a047-27a0b5e33793"> | <img width="1011" alt="Screenshot 2024-01-25 at 11 42 08 AM" src="https://github.com/TACC/Core-Styles/assets/63771558/0e5ec891-dc51-410f-9516-a89778d6f8f4"> |
| <img width="1011" alt="Screenshot 2024-01-25 at 11 43 43 AM" src="https://github.com/TACC/Core-Styles/assets/63771558/f354e236-88b5-44ae-a67a-97e243435981"> | <img width="1011" alt="Screenshot 2024-01-25 at 11 43 51 AM" src="https://github.com/TACC/Core-Styles/assets/63771558/ebcaf6ef-c92a-4cc1-957a-ea2a1b0050c4"> |
| <img width="1011" alt="Screenshot 2024-01-25 at 11 44 45 AM" src="https://github.com/TACC/Core-Styles/assets/63771558/c90e11be-0a06-4645-9268-badcb3fe56c2"> | <img width="1011" alt="Screenshot 2024-01-25 at 11 44 58 AM" src="https://github.com/TACC/Core-Styles/assets/63771558/60efd20d-e844-4fd3-a3b2-7463c9ed69f1"> |
| <img width="1011" alt="Screenshot 2024-01-25 at 11 46 24 AM" src="https://github.com/TACC/Core-Styles/assets/63771558/9107b729-8120-43e1-b47b-f7101e50c2ac"> | <img width="1011" alt="Screenshot 2024-01-25 at 11 46 34 AM" src="https://github.com/TACC/Core-Styles/assets/63771558/e603a5cc-74de-4c61-b234-7eed3f49fca9"> |
| <img width="1011" alt="Screenshot 2024-01-25 at 11 47 16 AM" src="https://github.com/TACC/Core-Styles/assets/63771558/141e01d8-684a-419d-b324-07322052cab6"> | <img width="1011" alt="Screenshot 2024-01-25 at 11 47 29 AM" src="https://github.com/TACC/Core-Styles/assets/63771558/f6972bdc-0fed-4539-bf96-1c16dbc5acc2"> |
